### PR TITLE
Ex2 sol & unit tests

### PR DESCRIPTION
**Pull request by noa-kogonia**

### CODE REVIEW

File: src/main/java/ex2/Ex2sol.kt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package ex2
+
+fun isCorrectCapitalization(word: String?): Boolean {
+    val regex = "(^[A-Z][A-Z]+)||(^[A-Z][a-z]+)||(^[a-z][a-z]+)"
```
Comment (af-fess) [inline src/main/java/ex2/Ex2sol.kt:4]: too complicated, we can simplify it. But definitely, regex is the best approach here.

**Review by af-fess — COMMENTED** [review]
Try to get rid of gradle7 folder in MR.

File: src/test/java/testsKotlin.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.MethodSource
+import java.util.stream.Stream
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class testsKotlin {
+
+     private fun wordAndExpectedProvider(): Stream<Arguments?>? {
+        return Stream.of(
+            Arguments.arguments("NOA", true),
```
[the remark this answers is not in the document]
Comment (noa-kogonia) [reply src/test/java/testsKotlin.kt:14]: according to the assignment:
2. A word may be all uppercase. “PEACHY”

File: src/main/java/ex2/Ex2sol.kt (hunk 1 above)
Comment (noa-kogonia) [reply src/main/java/ex2/Ex2sol.kt]: simplified a bit to val regex = "(^[A-Z]+)||(^[A-Z][a-z]+)||(^[a-z]+)"

**Pull request merged**